src/Models/Sitemap.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Rapidez\Sitemap\Models;
+
+use Illuminate\Support\Facades\Cache;
+use Rapidez\Core\Models\Model;
+use SimpleXMLElement;
+
+class Sitemap extends Model
+{
+    protected $table = 'sitemap';
+
+    public static function getCachedByStoreId(): ?array
+    {
+        $cacheKey = 'sitemaps.'.config('rapidez.store');
+
+        return Cache::rememberForever($cacheKey, function () {
+            return self::where('store_id', config('rapidez.store'))
+                ->get()
+                ->flatMap(fn ($sitemap) => self::getSitemapsFromIndex($sitemap->toArray()))
+                ->toArray();
+        });
+    }
+
+    protected static function getSitemapsFromIndex(array $sitemap): array
+    {
+        $sitemapUrl = config('rapidez.magento_url').$sitemap['sitemap_path'].$sitemap['sitemap_filename'];
```

Comment:
If we know the url, why do we need to load the content in? We can just reference it in our index right?

Reply:
It already is, like i said in the description https://github.com/rapidez/sitemap/pull/1#issue-2632172308

We have to check every sitemap from Magento to see if it is an XML containing just a `<urlset`, if so we add the url of this XML to our index.
The Magento XML can also contain a sitemap index, which would look something like this:

`<?xml version="1.0" encoding="UTF-8"?>
<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
<sitemap><loc>https://test.xyz.nl/media/sitemap/theme/sitemap-2-1.xml</loc><lastmod>2024-11-01T08:10:35+00:00</lastmod></sitemap>
<sitemap><loc>https://test.xyz.nl/media/sitemap/theme/sitemap-2-2.xml</loc><lastmod>2024-11-01T08:10:35+00:00</lastmod></sitemap>
<sitemap><loc>https://test.xyz.nl/media/sitemap/theme/sitemap-2-3.xml</loc><lastmod>2024-11-01T08:10:35+00:00</lastmod></sitemap>
</sitemapindex>`

In this case we extract each sitemap and add it to our own index.